app/src/main/java/com/douglasbruce/tasky/core/data/database/dao/EventDao.kt

```diff
@@ -12,5 +13,14 @@ interface EventDao {
             WHERE id = :eventId
         """,
     )
-    fun getEventById(eventId: String): EventEntity
+    fun getEventById(eventId: String): Flow<EventEntity>
+
+    @Query(
+        value = """
+            SELECT * FROM events
+            WHERE 'from' >= :date
+            AND 'to' < :date
+        """,
+    )
+    fun getEventsForDate(date: Long): Flow<List<EventEntity>>
```

Comment:
I don't fully get how this query gives you the events for the date (and why you're passing a range for all other agenda item types, but not for events). If an event goes from 23:30 to 00:30 and the current time would be 23:45, it doesn't seem this query would catch that event.